apps/backend/src/app/auth/service/keycloak-auth.service.ts

```diff
@@ -0,0 +1,232 @@
+import { Injectable, Logger, UnauthorizedException } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { ConfigService } from '@nestjs/config';
+import { firstValueFrom } from 'rxjs';
+
+export interface KeycloakTokenResponse {
+  access_token: string;
+  token_type: string;
+  expires_in: number;
+  refresh_token: string;
+  scope?: string;
+  id_token?: string;
+}
+
+export interface KeycloakUserInfo {
+  sub: string;
+  preferred_username: string;
+  given_name?: string;
+  family_name?: string;
+  email?: string;
+  realm_access?: {
+    roles: string[];
+  };
+}
+
+@Injectable()
+export class KeycloakAuthService {
+  private readonly logger = new Logger(KeycloakAuthService.name);
+
+  constructor(
+    private readonly httpService: HttpService,
+    private readonly configService: ConfigService
+  ) {}
+
+  /**
+   * Generate the Keycloak authorization URL for the Authorization Code flow
+   * @param state - Random state parameter for security
+   * @param redirectUri - Callback URL after authentication
+   * @returns Authorization URL
+   */
+  getAuthorizationUrl(state: string, redirectUri: string): string {
+    const keycloakUrl = this.configService.get<string>('KEYCLOAK_URL');
```

Comment:
Die 'keycloakXY'-Variablen werden in jeder Methode dieser Klasse neu gesetzt. Könnte man sie nicht einmalig beim initialisieren Klasse setzen?